packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js

```diff
@@ -483,112 +614,41 @@ const defaultTwoColumnModuleLayout = <T: { +[string]: mixed }>({
       positionCache,
     };
 
-    if (hasMultiColumnItems) {
-      // Currently we only support one two column item at the same time, more items will be supporped soon
-      const multiColumnIndex = itemsWithoutPositions.indexOf(multiColumnItems[0]);
-      const oneColumnItems = itemsWithoutPositions.filter(
-        (item) => !item.columnSpan || item.columnSpan === 1,
-      );
-
-      // The empty columns can be different from columnCount if there are
-      // items already positioned from previous batches
-      const emptyColumns = heights.reduce((acc, height) => (height === 0 ? acc + 1 : acc), 0);
-
-      const multiColumnItemColumnSpan = Math.min(
-        parseInt(multiColumnItems[0].columnSpan, 10),
-        columnCount,
-      );
-
-      // Skip the graph logic if the two column item can be displayed on the first row,
-      // this means graphBatch is empty and multi column item is positioned on its
-      // original position (twoColumnIndex)
-      const fitsFirstRow = emptyColumns >= multiColumnItemColumnSpan + multiColumnIndex;
-
-      // When multi column item is the last item of the first row but can't fit
-      // we need to fill those spaces with one col items
-      const replaceWithOneColItems = !fitsFirstRow && multiColumnIndex < emptyColumns;
-
-      // Calculate how many items are on pre array and how many on graphBatch
-      // pre items are positioned before the two column item
-      const splitIndex = calculateSplitIndex({
-        oneColumnItemsLength: oneColumnItems.length,
-        multiColumnIndex,
-        emptyColumns,
-        fitsFirstRow,
-        replaceWithOneColItems,
-      });
-
-      const pre = oneColumnItems.slice(0, splitIndex);
-      const graphBatch = fitsFirstRow
-        ? []
-        : oneColumnItems.slice(splitIndex, splitIndex + MULTI_COL_ITEMS_MEASURE_BATCH_SIZE);
-
-      // Get positions and heights for painted items
+    if (multiColumnItems.length > 0) {
       const { positions: paintedItemPositions, heights: paintedItemHeights } =
         getOneColumnItemPositions({
-          items: [...itemsWithPositions, ...pre],
+          items: itemsWithPositions,
           heights,
           ...commonGetPositionArgs,
         });
 
-      // Adding the extra prev column items to the position cache
-      if (paintedItemPositions.length > itemsWithPositions.length) {
-        paintedItemPositions.forEach(({ item, position }) => {
-          positionCache.set(item, position);
-        });
+      let currentPositions = paintedItemPositions;
+      let currentHeights = paintedItemHeights;
+
+      for (let i = 0; i < multiColumnItems.length; i += 1) {
```

Comment:
would it make sense to break these up into batches instead? i think a few things that are confusing here:
- we always filter the full set of items for each iteration of the loop
- _what_ actually gets filtered is dependent on the previous iteration placing more items in the position cache

instead, we could also do:
```
const batchSize = multiColumnItems.length;
const batches = Array.from({ length: batchSize }, () => ([])).map((batch, i) => {
    const startIndex = i === 0 ? 0 : items.indexOf(multiColumnItems[i]);
    const endIndex = i + 1 === multiColumnItems.length ? items.length : items.indexOf(multiColumnItems[i + 1])
    return items.slice(startIndex, endIndex);
});
const { heights: currentHeights, positions: currentPositions } = batches.reduce((acc, itemsToPosition) => {
  return getPositionsWithMultiColumnItem({
    ....
  })
}, { heights:paintedItemHeights, positions: paintedItemPositions });
```

unless i'm mistaken, this should yield the same result since each loop is still using the updated positions and heights from the previous batch. the only difference is that the dependency is a bit clearer (imo)